etc/scripts/get-stack.sh

```diff
@@ -748,11 +748,71 @@ on_path() {
 # Check whether ~/.local/bin is on the PATH, and print a warning if not.
 check_home_local_bin_on_path() {
   if ! on_path "$HOME_LOCAL_BIN" ; then
-    #TODO: offer to add it for the user (pull requests welcome!)
     info "WARNING: '$HOME_LOCAL_BIN' is not on your PATH."
     info "    Stack will place the binaries it builds in '$HOME_LOCAL_BIN' so"
     info "    for best results, please add it to the beginning of PATH in your profile."
     info ""
+    # detect which profile file to use, then print a message about updating it
+    if [ -n "$BASH_VERSION" ]; then
+      if [ -f "$HOME/.bash_profile" ]; then
+        profile_file="$HOME/.bash_profile"
+      else
+        profile_file="$HOME/.bashrc"
+      fi
+    elif [ -n "$ZSH_VERSION" ]; then
+      profile_file="$HOME/.zshrc"
+    elif [ -n "$FISH_VERSION" ]; then
+      profile_file="$HOME/.config/fish/config.fish"
+    elif [ -n "$PROFILE" ]; then
+      profile_file="$PROFILE"
+    elif [ -f "$HOME/.bash_profile" ]; then
+      profile_file="$HOME/.bash_profile"
+    elif [ -f "$HOME/.bashrc" ]; then
+      profile_file="$HOME/.bashrc"
+    elif [ -f "$HOME/.zshrc" ]; then
+      profile_file="$HOME/.zshrc"
+    elif [ -f "$HOME/.config/fish/config.fish" ]; then
+      profile_file="$HOME/.config/fish/config.fish"
+    elif [ -f "$HOME/.profile" ]; then
+      profile_file="$HOME/.profile"
+    else
+      info "    (profile not found; please add it to your PATH manually)"
+    fi
+
+    # print a message about updating profile (if found)
+    if [ -n "$profile_file" ]; then
+      info "    You can do this by running the following command:"
+      info "    echo 'export PATH=\"$HOME_LOCAL_BIN:\$PATH\"' >> \"$profile_file\""
+      info "    (You may need to restart your shell for this to take effect.)"
+      info ""
+    
+      # prompt to update it on their behalf, unless QUIET is set.
+      if [ -z "$QUIET" ]; then
+        while true; do
+          # default to no.
+          yesno=n
+          info "Would you like this installer to add it to your PATH in '$profile_file'?"
+          info "    (this will be done by adding export PATH=\"$HOME_LOCAL_BIN:\$PATH\" to it)"
+          info "    (you may need to restart your shell for this to take effect)"
+          info "    [y] Yes, prepend  [n] No (default)"
+          read -p "" yesno
```

Comment:
I think this mechanism, together with line 793, does not implement a default value of `n` in variable `yesno` if ENTER is pressed. I think you may need something like:
~~~bash
read -p "" yesno
: ${yesno:=n}
~~~
~~~

Reply:
thanks for testing on windows! i don't have access to one of those.
fix incoming.

Reply:
handled. also added a timeout in case someone uses this script in a docker image or something like that and does not set QUIET.

Reply:
I think timeout will be important because this script is used in CI, historically it has not needed any human intervention, its `-q` flag is not documented online, and (in any event) `-q` (quiet) does not always imply 'non-interactive'.

Consequently, further, I wonder if it should also skip the prompt if any of the environment variables that indicate popular CI environments are present. ChatGPT offers up the following list (which may need to be checked for hallucinations):
~~~text
Jenkins: JENKINS_HOME
Travis CI: TRAVIS
CircleCI: CIRCLECI
GitHub Actions: GITHUB_ACTIONS
GitLab CI: GITLAB_CI
Azure Pipelines: TF_BUILD
Bitbucket Pipelines: BITBUCKET_BUILD_NUMBER
~~~

Reply:
The `-q` flag is now documented online as 'suppresses output and specifies non-intervention (likely a prerequisite for the use of the script in CI environments)'.

Reply:
agreed that timeout is important. hm. do you think this script needs to cover for all those, or just document that the `QUIET` env should be set in any of those situations?

Reply:
I think that something like:
~~~sh
if [ -z "$QUIET" ] || ! has_ci_environment ; then
~~~

would assist because I am concerned that some existing use of the script in CI may simply assume that it is non-interactive.

Reply:
I read some more about environment variables in CI and added a commit, accordingly.